src/client-agent/config.c

```diff
@@ -128,14 +128,26 @@ cJSON *getLabelsConfig(void) {
 
     unsigned int i;
     cJSON *root = cJSON_CreateObject();
-    cJSON *labels = cJSON_CreateObject();
+    cJSON *label_s = cJSON_CreateArray();
+    cJSON *label = cJSON_CreateObject();
+    cJSON *labels = cJSON_CreateArray();
 
     if (agt->labels) {
         for (i=0;agt->labels[i].key;i++) {
-            cJSON_AddStringToObject(labels,agt->labels[i].key,agt->labels[i].value);
+            cJSON *label_k = cJSON_CreateObject();
+            cJSON_AddStringToObject(label_k,"value", agt->labels[i].value);
+            cJSON_AddStringToObject(label_k,"key", agt->labels[i].key);
+            if (agt->labels[i].flags.hidden)
+                cJSON_AddStringToObject(label_k,"hidden", "yes");
+            if (agt->labels[i].flags.system)
+                cJSON_AddStringToObject(label_k,"system", "yes");
+            cJSON_AddItemToObject(label_s,"",label_k);
+            free(label_k);
```

Comment:
Objects added to another object using `cJSON_AddItemToObject` must not be freed, as JSON_Delete will take care of freeing them. 
That `free` breaks the parent object.